### PR TITLE
added metric "rocksdb_total_sst_files"

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/19033
added metric "rocksdb_total_sst_files" to count the number of sst files, aggregated over all levels of the LSM tree.

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.11: https://github.com/arangodb/arangodb/pull/19034
  - [x] Backport for 3.10: this PR
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: tests/js/client/shell/shell-total-sst-files-metric-noncluster.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      let res = arango.GET_RAW("/_admin/metrics");
+      assertEqual(200, res.code);
+
+      const extract = (text, name) => {
+        let re = new RegExp("^" + name);
+        let matches = text.split('\n').filter((line) => !line.match(/^#/)).filter((line) => line.match(re));
+        if (!matches.length) {
+          throw "Metric " + name + " not found";
+        }
+        return Number(matches[0].replace(/^.*?\s*([0-9.]+)$/, "$1"));
```
Comment (mpoeter) [inline tests/js/client/shell/shell-total-sst-files-metric-noncluster.js:64]: This regex is different from the devel PR. Is this intentional?

**Review by mpoeter — APPROVED** [review]

File: tests/js/client/shell/shell-total-sst-files-metric-noncluster.js (hunk 1 above)
Comment (mpoeter) [inline tests/js/client/shell/shell-total-sst-files-metric-noncluster.js:64]: This regex is different from the devel PR. Is this intentional?

Comment (jsteemann) [reply tests/js/client/shell/shell-total-sst-files-metric-noncluster.js:64]: Yes. The regex in 3.10 is broken.

Comment (jsteemann) [reply tests/js/client/shell/shell-total-sst-files-metric-noncluster.js:64]: Yes, the regex in 3.10 is broken.

**Review by maierlars — APPROVED** [review]

**Pull request merged**